### DISCUSSION TITLE
Use get-ports for local server ports

### DISCUSSION THREAD

**Discussion by gmaclennan** [digidem/mapeo-desktop#552] [Ideas]
**Describe the idea**
Check for port availability before listening on local servers

**Additional context**
In [mapfilter-desktop](https://github.com/digidem/mapfilter-desktop/blob/master/main/app.js#L71) we use [get-ports](https://www.npmjs.com/package/get-ports) to find an available local port, this avoids potential conflict with existing in-use ports.

**Comment by okdistribute** [comment]
This should only be used for mapeo-server, not osm-p2p-server because custom imagery in iD editor depends upon the port being static (persistent) between app restarts.

**Comment by gmaclennan** [comment]
Good point, we should change the way custom imagery is handled at some point, but in the meantime this is good to bear in mind.
